Makefile

```diff
@@ -103,12 +94,18 @@ ifdef RELEASE
 DEFINES += -DNO_ASSERT -DRELEASE
 endif
 
-LATEST_RELEASE=$(shell git tag | grep RELEASE_ | sort | tail -1)
-# use egrep to count lines instead of wc to avoid whitespace error on Mac
-COMMITS_SINCE_RELEASE=$(shell git log --oneline $(LATEST_RELEASE)..HEAD | egrep -c .)
-ifneq ($(COMMITS_SINCE_RELEASE),0)
-DEFINES += -DBUILDNUMBER=\"$(COMMITS_SINCE_RELEASE)\"
+LATEST_RELEASE = $(shell egrep "define JS_VERSION .*\"$$" src/jsutils.h | egrep -o '[0-9]v[0-9]+')
+ifdef RELEASE_LABEL
```

Comment:
So I changed LATEST_RELEASE because it didn't work due to the fact that travis does a shallow clone and the release tags are too far back. Since what Espruino prints out at boot comes from a #define in jsutil.h I figured the Makefile might as well get it from there too.

If RELEASE_LABEL is not defined, which is the default, then the SUB_RELEASE is exactly what it was before, but it still suffers from the problem that the travis clone may not have the release tag. This is the case currently. My vote would be to switch from counting commits to putting the date in there, something like YYYYMMDD_HHMM. Alternatively we could use date and SHA, like YYYYMMDD_SSSSSS where the S's are the first 7 letters of the commit SHA.

If RELEASE_LABEL is defined, which is not default, then you get $(RELEASE_LABEL)_$(BRANCH_NAME). I set RELEASE_LABEL to "tve" in a travis env variable on my repository, this way a build of mine comes out saying something like "v1.81_tve_unify". I'll add the first few letters of the commit SHA, I think.